bibtexparser/bibdatabase.py

```diff
@@ -66,6 +67,23 @@ def __init__(self):
         #: List of fields that should not be updated when resolving crossrefs
         self._not_updated_by_crossref = ['_FROM_CROSSREF']
 
+    def __eq__(self, other: Any) -> bool:
+        """Equality == method."""
+        if (
```

Comment:
I guess this can be simplified?
```
        return (
            isinstance(other, BibDatabase)
            and other.entries == self.entries
            and other.comments == self.comments
            and other.strings == self.strings
            and other.preambles == self.preambles
        )
```

Reply:
Something like `self.__dict__ == other.__dict__`? Personally, I prefer the readable and explicit version when it's only a few attributes.

Reply:
self.__dict__ == other.__dict_ has the advantage that the implementation does not have to change as new attributes are added to a class.

However, this is not the point I wanted to make ;-) I'm fine with explicitly listing attributes, **but** instead of `if (condition) then return True else return False`, just use `return condition`; as shown in the example in my comment.

Reply:
Somehow I missed the example in your comment, sure that works.

Reply:
Ended up using `.__dict__` as it's simpler since some attributes are not defined in `__init__` but by other methods.